### PR TITLE
feat(pagination): add disabled variation

### LINKED ISSUE #1963: Add example for disabled pagination
Once https://github.com/patternfly/patternfly-next/issues/1962 is completed, let's add an example in the pagination component of a disabled state with the options menu and form fields disabled.

Related to https://github.com/patternfly/patternfly-next/issues/1785

cc @mcarrano

### PR DESCRIPTION
**Pull request by christiemolloy**
closes #1963 

Adds disabled variation for the top pagination component - assuming that we don't need to also show an example for the bottom variation?

### CODE REVIEW

**Comment by christiemolloy** [comment]
Thanks @mattnolting updated!!

**Comment by mcoker** [comment]
@christiemolloy looks like in https://github.com/patternfly/patternfly-next/pull/1973, we missed adding `disabled` to the toggle. Here's the discussion on it when we added the disabled variation of the dropdown component - https://github.com/patternfly/patternfly-next/pull/1512#pullrequestreview-210241111 and you'll find it on the dropdown's disabled example - https://pf4.patternfly.org/components/Dropdown/examples/.

And in this PR, we need to add `disabled` to `.pf-c-options-menu__toggle-button`, which is covered in the docs above, which are also here in the options menu docs

> When [`.pf-m-disabled`] is used, disabled should also be added to any form elements in .pf-c-options-menu__toggle

File: src/patternfly/components/Pagination/examples/pagination-top-disabled-example.hbs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{{#> pagination}}
+  {{#> pagination-total-items}}
+    <b>1 - 10</b> of <b>37</b>
+  {{/pagination-total-items}}
+
+  {{> pagination-options-menu id="pagination-options-menu-top-disabled-example"  options-menu--IsText="true" options-menu-toggle--IsDisabled="true"}}
+
+  {{#> pagination-nav}}
+    {{#> button button--modifier="pf-m-plain pf-m-disabled" button--attribute='aria-label="Go to first page" aria-disabled="true"'}}
```
Comment (mcoker) [inline src/patternfly/components/Pagination/examples/pagination-top-disabled-example.hbs]: Looks like in the pagination component, we're using `pf-m-disabled` wrong. We should be using `disabled` on `button`s and `pf-m-disabled` on `a` elements. So these buttons need to be updated to use `disabled`

Do you want to do that as part of this issue, or a follow up issue?

Comment (christiemolloy) [reply src/patternfly/components/Pagination/examples/pagination-top-disabled-example.hbs]: I can do it in this issue

**Comment by christiemolloy** [comment]
@mcoker updated... Added disabled variation to the plain options menu with text. I also added `disabled` to both `pf-c-options-menu__toggle` and `pf-c-options-menu__toggle-button` .. In the dropdown component I don't see the `pf-m-disabled` modifier on the buttons, I only see `disabled`, so i'm a little confused as to what to do there.

Screen Shot 2019-07-08 at 10 45 44 AM

File: src/patternfly/components/OptionsMenu/docs/code.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
 | -- | -- | -- |
```
Comment (christiemolloy) [inline src/patternfly/components/OptionsMenu/docs/code.md]: @mcoker @jgiardino now that the react implementation is complete should I create an issue to fill in the options menu docs? or do you know if this is still on hold?

File: src/patternfly/components/Pagination/examples/pagination-bottom-example.hbs [hunk 3]
```diff
@@ -2,10 +2,10 @@
   {{> pagination-options-menu id="pagination-options-menu-bottom-example"  options-menu--IsText="true"}}
 
   {{#> pagination-nav}}
-    {{#> button button--modifier="pf-m-plain pf-m-disabled" button--attribute='aria-label="Go to first page" aria-disabled="true"'}}
+    {{#> button button--modifier="pf-m-plain" button--attribute='disabled aria-label="Go to first page" aria-disabled="true"'}}
```
Comment (mcoker) [inline src/patternfly/components/Pagination/examples/pagination-bottom-example.hbs]: I don't think we need `aria-disabled`, either, if the element has the `disabled` attribute. @jgiardino can you confirm?

**Comment by mcoker** [comment]
On the "Options menu - disabled" and "Options menu - plain
" (disabled) examples, we don't need `.pf-m-disabled` on the toggle since it's a `<button>`.

File: src/patternfly/components/Pagination/examples/pagination-bottom-example.hbs (hunk 3 above)
Comment (christiemolloy) [reply src/patternfly/components/Pagination/examples/pagination-bottom-example.hbs]: In this article it says: "A button with disabled as well as the button with aria-disabled will be labeled as "dimmed button" https://a11y-101.com/development/aria-disabled

Comment (christiemolloy) [reply src/patternfly/components/Pagination/examples/pagination-bottom-example.hbs]: Looks like we only need "disabled"

**Comment by christiemolloy** [comment]
thanks @mcoker updated!

File: src/patternfly/components/OptionsMenu/options-menu.scss [hunk 4]
```diff
@@ -174,9 +174,9 @@
   &:disabled {
     pointer-events: none;
 
-    &:not(.pf-m-plain) {
```
Comment (mcoker) [inline src/patternfly/components/OptionsMenu/options-menu.scss]: Removing this gives a background to the plain (non `.pf-m-text`) button.

Screen Shot 2019-07-09 at 10 32 16 AM

I also don't know that you need the `&:not(.pf-m-plain)` on L179, either. It's correct code-wise, but seems unnecessary and increases specificity.

What if you rewrote this block to


```
  &.pf-m-disabled,
  &:disabled {
    pointer-events: none;

    &:not(.pf-m-plain),
    &.pf-m-text {
      --pf-c-options-menu__toggle--Background: var(--pf-c-options-menu__toggle--disabled--BackgroundColor);
    }

    &::before {
      border: 0;
    }
  }
```

**Comment by christiemolloy** [comment]
I'm good with that change! updated! @mcoker

File: src/patternfly/components/OptionsMenu/docs/code.md [hunk 5]
```diff
@@ -6,7 +6,7 @@
...
 | `role` or `aria` | `pf-c-options-menu` |  accessibility notes. |
 *Note:* The attribute `aria-selected="true"` should be set programmatically to the selected item(s).
-
+| `disabled` | `.pf-c-options-menu__toggle`, `.pf-c-options-menu__toggle-button` | Disables the options menu toggle and toggle button and removes it from keyboard focus. |
```
Comment (mcoker) [inline src/patternfly/components/OptionsMenu/docs/code.md]: I think you want to move this above the `*Note:*`

**Pull request merged**